### ISSUE TITLE
I start a new project SRS, to delivery in RTMP/HLS

### ISSUE THREAD

**Issue by winlinvip** [arut/nginx-rtmp-module#626]
Hi, guys, nginx-rtmp is a great project, but seems not update for a long time.
I start a new project SRS, to delivery in RTMP/HLS and HTTP FLV, and other features.
The project: https://github.com/simple-rtmp-server/srs
And now I am deving on 2.0: https://github.com/simple-rtmp-server/srs/tree/2.0release

**Comment by winlinvip** [comment]
My contact: 
https://github.com/simple-rtmp-server/srs/wiki/v1_CN_Contact
https://github.com/simple-rtmp-server/srs/wiki/v1_EN_Contact

Skype: winterserver

**Comment by winlinvip** [comment]
Why SRS? https://github.com/simple-rtmp-server/srs/tree/2.0release#why-srs

Completely rewrite HLS following m3u8/ts spec, and HLS support h.264+aac/mp3.
High efficient RTMP deliverying support 7k+ concurrency, vhost based, both origin and edge.
Embeded simplified media HTTP server for HLS, api and HTTP flv/ts/mp3/aac streaming.
Variety input: RTMP, pull by ingest file or stream(HTTP/RTMP/RTSP), push by stream caster RTSP/MPEGTS-over-UDP.
Popular internet delivery: RTMP/HDS for flash, HLS for mobile(IOS/IPad/MAC/Android), HTTP flv/ts/mp3/aac streaming for user prefered.
Enhanced DVR and hstrs: segment/session/append plan, customer path and HTTP callback. the hstrs(http stream trigger rtmp source) enable the http-flv stream standby util encoder start publish, similar to rtmp, which will trigger edge to fetch from origin.
Multiple feature: transcode, forward, ingest, http hooks, dvr, hls, rtsp, http streaming, http api, refer, log, bandwith test and srs-librtmp.
Best maintainess: simple arch over state-threads(coroutine), single thread, single process and for linux/osx platform, common server x86-64/i386/arm/mips cpus, rich comments, strictly follows RTMP/HLS/RTSP spec.
Easy to use: both English and Chinese wiki, typically config files in trunk/conf, traceable and session based log, linux service script and install script.
MIT license, open source with product management and evolution.
Enjoy it!

**Comment by jalonsoa** [comment]
Hi Winlin,
  you proyect looks fine, but ... I think that HDS is a must .. you will
support it or not ?

  https://github.com/simple-rtmp-server/srs/wiki/v1_EN_DeliveryHLS

j

**Comment by winlinvip** [comment]
Yep, the input of SRS is RTMP(and we provides the ingester and stream caster), and the output or delivery in RTMP/HTTP-FLV/HLS/HDS.
https://github.com/simple-rtmp-server/srs/tree/2.0release#stream-architecture

**Comment by winlinvip** [comment]
I compared SRS with others media servers, please read https://github.com/simple-rtmp-server/srs/tree/2.0release#compare



## Compare

Compare SRS with other media server.

### Stream Delivery

|   Feature     |   SRS     |   NGINX   | CRTMPD    | FMS       |   WOWZA   |
|   ----------- |   ------- |   -----   | --------- | --------  |   ------  |
|   RTMP        |   Stable  |   Stable  |   Stable  |   Stable  |   Stable  |
|   HLS         |   Stable  |   Stable  |   X       |   Stable  |   Stable  |
|   HDS         |   Stable  |   X       |   X       |   Stable  |   Stable  |
|   HTTP FLV    |   Stable  |   X       |   X       |   X       |   X       |
|   HLS(aonly)  |   Stable  |   X       |   X       |   Stable  |   Stable  |
|   HTTP Server |   Stable  |   Stable  |   X       |   X       |   Stable  |

### Cluster

|   Feature     |   SRS     |   NGINX   | CRTMPD    | FMS       |   WOWZA   |
|   ----------- |   ------- |   -----   | --------- | --------  |   ------  |
|   RTMP Edge   |   Stable  |   X       |   X       |   Stable  |   X       |
|   RTMP Backup |   Stable  |   X       |   X       |   X       |   X       |
|   VHOST       |   Stable  |   X       |   X       |   Stable  |   Stable  |
|   Reload      |   Stable  |   Stable  |   X       |   X       |   X       |
|   Forward     |   Stable  |   X       |   X       |   X       |   X       |
|   ATC         |   Stable  |   X       |   X       |   X       |   X       |

### Stream Service

|   Feature     |   SRS     |   NGINX   | CRTMPD    | FMS       |   WOWZA   |
|   ----------- |   ------- |   -----   | --------- | --------  |   ------  |
|   DVR         |   Stable  |   Stable  |   X       |   X       |   Stable  |
|   Transcode   |   Stable  |   X       |   X       |   X       |   Stable  |
|   HTTP API    |   Stable  |   Stable  |   X       |   X       |   Stable  |
|   HTTP hooks  |   Stable  |   X       |   X       |   X       |   X       |
|   GopCache    |   Stable  |   X       |   X       |   Stable  |   X       |
|   Security    |   Stable  |   Stable  |   X       |   X       |   Stable  |
| Token Traverse|   Stable  |   X       |   X       |   Stable  |   X       |

### Performance

|   Feature     |   SRS     |   NGINX   | CRTMPD    | FMS       |   WOWZA   |
|   ----------- |   ------- |   -----   | --------- | --------  |   ------  |
|   Concurrency |   7.5k    |   3k      |   2k      |   2k      |   3k      |
|   RTMP Latency|   0.1s    |   3s      |   3s      |   3s      |   3s      |
|   HLS Latency |   10s     |   30s     |   X       |   30s     |   30s     |

### Stream Caster

|   Feature     |   SRS     |   NGINX   | CRTMPD    | FMS       |   WOWZA   |
|   ----------- |   ------- |   -----   | --------- | --------  |   ------  |
|   Ingest      |   Stable  |   X       |   X       |   X       |   X       |
|   Push MPEGTS | Experiment|   X       |   X       |   X       |   Stable  |
|   Push RTSP   | Experiment|   X       |   Stable  |   X       |   Stable  |
| Push HTTP FLV | Experiment|   X       |   X       |   X       |   X       |

### Debug System

|   Feature     |   SRS     |   NGINX   | CRTMPD    | FMS       |   WOWZA   |
|   ----------- |   ------- |   -----   | --------- | --------  |   ------  |
|   BW check    |   Stable  |   X       |   X       |   X       |   X       |
| Tracable Log  |   Stable  |   X       |   X       |   X       |   X       |

### Docs

|   Feature     |   SRS     |   NGINX   | CRTMPD    | FMS       |   WOWZA   |
|   ----------- |   ------- |   -----   | --------- | --------  |   ------  |
|   Demos       |   Stable  |   X       |   X       |   X       |   X       |
|   WIKI(EN+CN) |   Stable  |  EN only  |   X       |   X       |   Stable  |

### Others 

|   Feature     |   SRS     |   NGINX   | CRTMPD    | FMS       |   WOWZA   |
|   ----------- |   ------- |   -----   | --------- | --------  |   ------  |
|   ARM/MIPS    |   Stable  |   Stable  |   X       |   X       |   X       |
| Client Library|   Stable  |   X       |   X       |   X       |   X       |

Remark:

1. Concurrency: The concurrency of single process.
1. HLS aonly: The HLS audio only streaming delivery.
1. BW check: The bandwidth check.
1. Security: To allow or deny stream publish or play.

**Comment by misiek08** [comment]
Still can't belive people didn't know about SRS :(
23 maj 2015 10:04 "winlin" <notifications@github.com> napisał(a):

**Comment by smicroz** [comment]
I will try :)

**Comment by grinningorange** [comment]
Seems like awesome !

**Comment by yverbin** [comment]
I try srs in different scenarios. Good range of options, many interesting moments, but the  logic of the live transcoding and generating chunks/playlists not very well.  
Also, about gopcache -  try to stream static image using fmle/camtwist :-))

**Comment by winlinvip** [comment]
Please file issues to SRS for any bug, suggestion and ideas: https://github.com/simple-rtmp-server/srs/issues/new
Or contact me by Skype: winterserver
Thanks~

**Comment by winlinvip** [comment]
SRS2 released alpha2: https://github.com/simple-rtmp-server/srs/releases/tag/2.0a1
And SRS3 supported nginx-rtmp style [exec](https://github.com/simple-rtmp-server/srs/wiki/v3_EN_NgExec) and [dvr api](https://github.com/simple-rtmp-server/srs/wiki/v3_EN_HTTPApi#raw-dvr).
Welcome to migrate from great nginx-rtmp to great SRS~
:)
